controller/src/main/java/org/jboss/as/controller/SimpleOperationDefinitionBuilder.java

```diff
@@ -23,17 +24,57 @@
  */
 public class SimpleOperationDefinitionBuilder {
 
-    private static AttributeDefinition[] NO_ATTRIBUTES = new AttributeDefinition[0];
+    private static final AttributeDefinition[] NO_ATTRIBUTES = new AttributeDefinition[0];
 
     public static SimpleOperationDefinitionBuilder of(String name, ResourceDescriptionResolver resolver) {
         return new SimpleOperationDefinitionBuilder(name, resolver);
     }
 
+    /**
+     * Convenience method that delegates to {@link #of(String, SimpleOperationDefinition)}.
+     * @param name the operation name
+     * @param basis an operation used to configure a new operation
+     * @return a new operation definition builder
+     * @throws IllegalArgumentException if the specified basis is not a {@link SimpleOperationDefinition}.
+     */
+    public static SimpleOperationDefinitionBuilder of(String name, OperationDefinition basis) {
+        if (!(basis instanceof SimpleOperationDefinition)) {
+            throw new IllegalArgumentException();
+        }
+        return of(name, (SimpleOperationDefinition) basis);
+    }
+
+    /**
+     * Creates an operation definition builder with the specified name, based on the configuration of the specified operation definition.
+     * @param name the operation name
+     * @param basis an operation used to configure a new operation
+     * @return a new operation definition builder
+     */
+    public static SimpleOperationDefinitionBuilder of(String name, SimpleOperationDefinition basis) {
+        SimpleOperationDefinitionBuilder builder = new SimpleOperationDefinitionBuilder(name, basis.getResolver())
+                .setAccessConstraints(basis.getAccessConstraints().toArray(AccessConstraintDefinition[]::new))
+                .setAttributeResolver(basis.getAttributeResolver())
+                .setDescriptionProvider(basis.getDescriptionProvider())
+                .setEntryType(basis.getEntryType())
+                .setParameters(basis.getParameters())
+                .setReplyParameters(basis.getReplyParameters())
+                .setReplyType(basis.getReplyType())
+                .setReplyValueType(basis.getReplyValueType())
+                .setStability(basis.getStability())
+                .withFlags(basis.getFlags())
+                ;
+        DeprecationData deprecation = basis.getDeprecationData();
+        if (deprecation != null) {
+            builder.setDeprecated(deprecation.getSince(), deprecation.isNotificationUseful());
+        }
+        return builder;
```

Comment:
It looks like this would also require the initialization of `replyAllowNull` from the `SimpleOperationDefinition` passed as argument.